keps/sig-api-machinery/2144-clientgo-apply/README.md

```diff
@@ -159,69 +159,48 @@ take an aligned approach to adding apply to clients in a typesafe way.
 The client-go typed clients will be extended to include Apply functions, e.g.:
 
 ```go
-func (c *deployments) Apply(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
-func (c *deployments) ApplyStatus(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
-func (c *deployments) ApplyScale(ctx Context, fieldManager string, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) Apply(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) ApplyStatus(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
+func (c *deployments) ApplyScale(ctx Context, deployment *appsv1apply.Deployment, opts metav1.ApplyOptions) (*Deployment, error)
 ```
 
 `ApplyOptions` will be added to metav1 even though `PatchOptions` will continue
-to be used over the wire. This will make it obvious in the Apply function
-signature that `fieldManager` is required.
+to be used over the wire:
 
 ```go
 type ApplyOptions struct {
   DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,1,rep,name=dryRun"`
```

Comment:
Everything looks good except that I don't understand how someone would turn on dry run? The NewApplyOptions function isn't much more convenient than just constructing the options object yourself?

Reply:
> Everything looks good except that I don't understand how someone would turn on dry run? The NewApplyOptions function isn't much more convenient than just constructing the options object yourself?

Maybe we don't need the constructor at all? By adding ApplyOptions we can also improve the godoc to make it clear what fields are required and also switch `Force` from a `*bool` to a `bool` and document when it should be set to true. Maybe that would be sufficient? I'll update the PR to suggest this way.

Reply:
I don't object to this as it is, but I'll note that we can no longer catch errors (missing field manager, lack of choice for force) at compile time.

Reply:
ack. That's the main thing we are losing here by favoring consistency across the method functions. I think I can live with it.